tests/system/feature_store/test_spark_engine.py

```diff
@@ -1146,3 +1153,387 @@ def test_ingest_with_steps_extractor(self, timestamp_col):
         )
         csv_path_storey = measurements.get_target_path(name="csv")
         read_and_assert(csv_path_spark, csv_path_storey)
+
+    @pytest.mark.parametrize("with_indexes", [True, False])
+    @pytest.mark.parametrize("join_type", ["inner", "outer"])
+    def test_relation_join(self, join_type, with_indexes):
```

Comment:
IIRC, there's a very similar test for storey engine - can't the same test be reused? It may be interesting to actually run the same test over different engines and verify that the same results are obtained.

Reply:
actually it exactly the same test so we can determined that the output of all the engines are the same.